bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/handler/SiemensHvacBridgeBaseThingHandler.java

```diff
@@ -0,0 +1,164 @@
+    private @Nullable SiemensHvacDeviceDiscoveryService discoveryService;
+    private final @Nullable HttpClientFactory httpClientFactory;
+    private final SiemensHvacMetadataRegistry metaDataRegistry;
+    private @Nullable SiemensHvacBridgeConfig config;
+
+    public SiemensHvacBridgeBaseThingHandler(Bridge bridge, @Nullable NetworkAddressService networkAddressService,
+            @Nullable HttpClientFactory httpClientFactory, SiemensHvacMetadataRegistry metaDataRegistry) {
+        super(bridge);
+        SiemensHvacConnector lcConnector = null;
+        this.httpClientFactory = httpClientFactory;
+        this.metaDataRegistry = metaDataRegistry;
+
+        lcConnector = this.metaDataRegistry.getSiemensHvacConnector();
+        if (lcConnector != null) {
+            lcConnector.setSiemensHvacBridgeBaseThingHandler(this);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void dispose() {
+        metaDataRegistry.invalidate();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "@text/offline.waiting-bridge-initialization");
+
+        SiemensHvacBridgeConfig lcConfig = getConfigAs(SiemensHvacBridgeConfig.class);
+        String baseUrl = null;
+
+        baseUrl = lcConfig.baseUrl;
+
+        if (baseUrl.isEmpty()) {
+            logger.debug("@text/offline.error-gateway-init");
+            return;
+        }
+
+        if (!baseUrl.startsWith("http://") && !baseUrl.startsWith("https://")) {
+            baseUrl = "http://" + baseUrl;
+        }
+
+        if (!baseUrl.endsWith("/")) {
+            baseUrl = baseUrl + "/";
+        }
+
+        config = lcConfig;
+
+        // Will read metadata in background to not block initialize for a long period !
+        scheduler.schedule(this::initializeCode, 1, TimeUnit.SECONDS);
+    }
+
+    public static String getStackTrace(final Throwable throwable) {
+        StringBuffer sb = new StringBuffer();
+
+        Throwable current = throwable;
+        while (current != null) {
+            sb.append(current.getLocalizedMessage());
+            sb.append(",\r\n");
+
+            Throwable cause = throwable.getCause();
+            if (cause != null) {
+                if (!cause.equals(throwable)) {
+                    current = current.getCause();
+                } else {
+                    current = null;
+                }
+            } else {
+                current = null;
+            }
+        }
+        return sb.toString();
+    }
+
+    private void initializeCode() {
+        try {
+            metaDataRegistry.readMeta();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (SiemensHvacException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("@text/offline.error-gateway-init", ex.getMessage()));
```

Comment:
```suggestion
                    "@text/offline.error-gateway-init[\"" + ex.getMessage() + "\"]");
```

See corresponding comment for the I18N text.

Reply:
fix on next commit